csharp/ql/lib/semmle/code/csharp/dataflow/ExternalFlow.qll

```diff
@@ -348,6 +348,24 @@ module CsvValidation {
         msg = "Invalid boolean \"" + b + "\" in " + pred + " model."
       )
     )
+    or
+    exists(string row, string kind | summaryModel(row) |
+      kind = row.splitAt(";", 8) and
+      not kind = ["taint", "value"] and
+      msg = "Invalid kind \"" + kind + "\" in summary model."
+    )
+    or
+    exists(string row, string kind | sinkModel(row) |
+      kind = row.splitAt(";", 7) and
+      not kind = ["code", "sql", "xss", "remote", "html"] and
+      msg = "Invalid kind \"" + kind + "\" in sink model."
+    )
+    or
+    exists(string row, string kind | sourceModel(row) |
+      kind = row.splitAt(";", 7) and
+      not kind = "local" and
+      msg = "Invalid kind \"" + kind + "\" in sink model."
```

Comment:
```suggestion
      msg = "Invalid kind \"" + kind + "\" in source model."
```
Looks like a copy and paste error.

The index in the `kind = row.split` stuff is the same for source and sink models?

Reply:
@intrigus-lgtm : Thank you very much for catching this!